File: baekjoon/js/30802.js
const fs = require("fs");
const buf = fs.readFileSync(0, "utf-8").toString().trim();

const inputs = buf.split("\n");
const n = Number(inputs[0]);
const size = inputs[1].split(" ").map(Number);
const [t, p] = inputs[2].split(" ").map(Number);

const tCount = size.reduce((acc, cur) => acc + Math.ceil(cur / t), 0);
const pCount = [Math.floor(n / p), n % p];

console.log(`${tCount}\n${pCount.join(" ")}`);

File: baekjoon/rust/30802.rs
use std::io::{read_to_string, stdin};

fn main() {
    let buf = read_to_string(stdin()).unwrap();
    let mut inputs = buf.trim().lines();

    let n = inputs.next().unwrap().parse::<u32>().unwrap();
    let size = inputs
        .next()
        .unwrap()
        .split_whitespace()
        .flat_map(str::parse::<u32>)
        .collect::<Vec<_>>();
    let bundle = inputs
        .next()
        .unwrap()
        .split_whitespace()
        .flat_map(str::parse::<u32>)
        .collect::<Vec<_>>();
    let [t, p] = bundle[..] else {
        panic!("wrong");
    };

    let t_count = size.iter().fold(0, |acc, cur| acc + cur.div_ceil(t));
    let p_count = (n / p, n % p);

    println!("{}", t_count);
    print!("{} {}", p_count.0, p_count.1);
}
